toolchain/diagnostics/sorting_diagnostic_consumer.h

```diff
@@ -35,10 +35,12 @@ class SortingDiagnosticConsumer : public DiagnosticConsumer {
   void Flush() override {
     llvm::stable_sort(diagnostics_,
                       [](const Diagnostic& lhs, const Diagnostic& rhs) {
-                        return std::tie(lhs.message.location.line_number,
-                                        lhs.message.location.column_number) <
-                               std::tie(rhs.message.location.line_number,
-                                        rhs.message.location.column_number);
+                        const auto& lhs_loc = lhs.messages[0].location;
+                        const auto& rhs_loc = rhs.messages[0].location;
```

Comment:
Should we still track the primary diagnostic index in the vector, so we can use it for sorting?

Reply:
I think this *is* the value we want for sorting, isn't it?

Reply:
Or put differently, we could iterate through both messages for comparing: I'm being lazy. But I believe the primary location is the first location.

Reply:
Hm, if we're just using prefixes of the primary diagnostic for "imported here" information, then yes, I suppose it is what we want for sorting.